### DISCUSSION TITLE
typechecking?

### DISCUSSION THREAD

**Discussion by dcsan** [supabase/supabase#337] [Questions]
is there any way to get some typechecking or inference/intellisense on the queries?
It seems the queries are just plain text strings?

I guess prisma is the best example nowadays in JS land, but could queries at least be as JSON objects, a bit like mongo queries?
https://github.com/crcn/sift.js#readme

**Comment by kiwicopple** [comment]
Hey @dcsan thanks for dropping in. This is definitely something we'll build, hopfully for a few languages beside JS too.

As you say, the direct queries are just JS strings (which can still have intellisense), but we will also work on native objects/entities (likely with some sort of state/store) to make development easier
